### PR TITLE
Add tvos as deployment target

### PR DESCRIPTION
**Pull request by hamchapman**

### CODE REVIEW

**Comment by hamchapman** [comment]
Think the `travis.yml` needs some looking at. Plus maybe creating some new scheme(s) / target(s) for tvOS

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 language: objective-c
-osx_image: xcode6.4
+osx_image: xcode7.2
 before_script:
     - bundle install
 script:
     - xcodebuild -project SocketRocket.xcodeproj -scheme "SocketRocket" -sdk iphonesimulator -configuration Debug -PBXBuildsContinueAfterErrors=0 ACTIVE_ARCH_ONLY=0 build test
     - xcodebuild -project SocketRocket.xcodeproj -scheme "SocketRocket-iOS" -sdk iphonesimulator -configuration Debug -PBXBuildsContinueAfterErrors=0 ACTIVE_ARCH_ONLY=0 build
-    - xcodebuild -project SocketRocket.xcodeproj -scheme "SocketRocketOSX" -sdk macosx10.10 -configuration Debug -destination "platform=OS X" -PBXBuildsContinueAfterErrors=0 build
+    - xcodebuild -project SocketRocket.xcodeproj -scheme "SocketRocketOSX" -sdk macosx10.11 -configuration Debug -destination "platform=OS X" -PBXBuildsContinueAfterErrors=0 build
```
Comment (dfed) [inline .travis.yml]: why is this change necessary?

File: .travis.yml [hunk 2]
```diff
@@ -1,9 +1,9 @@
 language: objective-c
-osx_image: xcode6.4
+osx_image: xcode7.2
 before_script:
     - bundle install
 script:
```
Comment (dfed) [inline .travis.yml]: Please add a tvOS build to Travis's CI

**Comment by dfed** [comment]
We'll need this build to go green before we can merge. Feel free to take a look at #279, which got darned close before it was derailed by conflicts.

**Comment by hamchapman** [comment]
It's gone green on Travis, but as far as I can tell, that's really just signifying that the builds have succeeded, not that the tests have run (and then passed).

Speaking of which, the `make test` command doesn't work for me when I'm running it locally. 

Here's a sample of the error output:

```
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/usr/include/machine/types.h:37:2: error:
      architecture not supported
#error architecture not supported
 ^
In file included from /Users/Hami/Pusher/SocketRocket/SocketRocket/SocketRocket-Prefix.pch:22:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:11:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/usr/include/sys/types.h:79:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/usr/include/sys/_types.h:33:
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/usr/include/machine/_types.h:34:2: error:
      architecture not supported
#error architecture not supported
 ^
In file included from /Users/Hami/Pusher/SocketRocket/SocketRocket/SocketRocket-Prefix.pch:22:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:11:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/usr/include/sys/types.h:79:
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/usr/include/sys/_types.h:55:9: error:
      unknown type name '__int64_t'; did you mean '__int128_t'?
typedef __int64_t       __darwin_blkcnt_t;      /* total blocks */
        ^
note: '__int128_t' declared here
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/usr/include/sys/_types.h:56:9: error:
      unknown type name '__int32_t'; did you mean '__int128_t'?
typedef __int32_t       __darwin_blksize_t;     /* preferred block size */
        ^
note: '__int128_t' declared here
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/usr/include/sys/_types.h:57:9: error:
      unknown type name '__int32_t'; did you mean '__int128_t'?
typedef __int32_t       __darwin_dev_t;         /* dev_t */
        ^
note: '__int128_t' declared here
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/usr/include/sys/_types.h:60:9: error:
      unknown type name '__uint32_t'; did you mean '__uint128_t'?
typedef __uint32_t      __darwin_gid_t;         /* [???] process and group IDs */
        ^
note: '__uint128_t' declared here
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/usr/include/sys/_types.h:61:9: error:
      unknown type name '__uint32_t'; did you mean '__uint128_t'?
typedef __uint32_t      __darwin_id_t;          /* [XSI] pid_t, uid_t, or gid_t*/
        ^
note: '__uint128_t' declared here
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/usr/include/sys/_types.h:62:9: error:
      unknown type name '__uint64_t'; did you mean '__uint128_t'?
typedef __uint64_t      __darwin_ino64_t;       /* [???] Used for 64 bit inodes */
        ^
note: '__uint128_t' declared here
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/usr/include/sys/_types.h:68:9: error:
      unknown type name '__darwin_natural_t'
typedef __darwin_natural_t __darwin_mach_port_name_t; /* Used by mach */
        ^
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/usr/include/sys/_types.h:70:9: error:
      unknown type name '__uint16_t'; did you mean '__uint128_t'?
typedef __uint16_t      __darwin_mode_t;        /* [???] Some file attributes */
        ^
note: '__uint128_t' declared here
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/usr/include/sys/_types.h:71:9: error:
      unknown type name '__int64_t'; did you mean '__int128_t'?
typedef __int64_t       __darwin_off_t;         /* [???] Used for file sizes */
        ^
note: '__int128_t' declared here
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/usr/include/sys/_types.h:72:9: error:
      unknown type name '__int32_t'; did you mean '__int128_t'?
typedef __int32_t       __darwin_pid_t;         /* [???] process and group IDs */
        ^
note: '__int128_t' declared here
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/usr/include/sys/_types.h:73:9: error:
      unknown type name '__uint32_t'; did you mean '__uint128_t'?
typedef __uint32_t      __darwin_sigset_t;      /* [???] signal set */
        ^
note: '__uint128_t' declared here
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/usr/include/sys/_types.h:74:9: error:
      unknown type name '__int32_t'; did you mean '__int128_t'?
typedef __int32_t       __darwin_suseconds_t;   /* [???] microseconds */
        ^
note: '__int128_t' declared here
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/usr/include/sys/_types.h:75:9: error:
      unknown type name '__uint32_t'; did you mean '__uint128_t'?
typedef __uint32_t      __darwin_uid_t;         /* [???] user IDs */
        ^
note: '__uint128_t' declared here
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/usr/include/sys/_types.h:76:9: error:
      unknown type name '__uint32_t'; did you mean '__uint128_t'?
typedef __uint32_t      __darwin_useconds_t;    /* [???] microseconds */
        ^
note: '__uint128_t' declared here
In file included from /Users/Hami/Pusher/SocketRocket/SocketRocket/SocketRocket-Prefix.pch:22:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:11:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/usr/include/sys/types.h:81:
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/usr/include/machine/endian.h:37:2: error:
      architecture not supported
#error architecture not supported
 ^
In file included from /Users/Hami/Pusher/SocketRocket/SocketRocket/SocketRocket-Prefix.pch:22:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:11:
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/usr/include/sys/types.h:95:9: error:
      unknown type name 'u_int64_t'
typedef u_int64_t               u_quad_t;       /* quads */
        ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
Command /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang failed with exit code 1

** BUILD FAILED **
```

I've made a PR (https://github.com/square/SocketRocket/pull/330) that aims to address some of those problems. Maybe it's completely off-base but it seemed to get the tests _closer_ to working for me.

Out of interest, do the tests work for you when running them locally?

**Comment by hamchapman** [comment]
Just signed the contributor agreement

**Comment by hamchapman** [comment]
Anything else that needs to be done before this can be merged?

**Comment by rafiki270** [comment]
Any news on this pretty please? :)

**Comment by hamchapman** [comment]
It is still in a state where it can be merged without conflicts and so it's just in the hands of the those with push privileges :

**Comment by asabin** [comment]
really eager for this update, anyone we can bug?

**Comment by hamchapman** [comment]
@dfed anything I can help out with so we can get this merged?

File: .travis.yml [hunk 3]
```diff
@@ -1,9 +1,27 @@
 language: objective-c
-osx_image: xcode6.4
+osx_image: xcode7.2
+sudo: false
+xcode_project: SocketRocket.xcodeproj
+env:
```
Comment (dfed) [inline .travis.yml]: can you give some context on why the `env` param is necessary here?

File: .travis.yml [hunk 4]
```diff
@@ -1,9 +1,27 @@
 language: objective-c
-osx_image: xcode6.4
+osx_image: xcode7.2
+sudo: false
```
Comment (dfed) [inline .travis.yml]: Does this change current behavior? If so, why was it necessary?

File: .travis.yml [hunk 5]
```diff
@@ -1,9 +1,27 @@
...
 before_script:
     - bundle install
 script:
-    - xcodebuild -project SocketRocket.xcodeproj -scheme "SocketRocket" -sdk iphonesimulator -configuration Debug -PBXBuildsContinueAfterErrors=0 ACTIVE_ARCH_ONLY=0 build test
-    - xcodebuild -project SocketRocket.xcodeproj -scheme "SocketRocket-iOS" -sdk iphonesimulator -configuration Debug -PBXBuildsContinueAfterErrors=0 ACTIVE_ARCH_ONLY=0 build
-    - xcodebuild -project SocketRocket.xcodeproj -scheme "SocketRocketOSX" -sdk macosx10.10 -configuration Debug -destination "platform=OS X" -PBXBuildsContinueAfterErrors=0 build
-    - pod lib lint --verbose --fail-fast
+    - xcodebuild -version
+    - xcodebuild -scheme "$SCHEME" -sdk "$SDK" -destination "$DESTINATION" -configuration Debug -PBXBuildsContinueAfterErrors=0 ACTIVE_ARCH_ONLY=0 build test | xcpretty -tc
+    - pod lib lint --verbose --fail-fast
```
Comment (dfed) [inline .travis.yml]: non-blocking nit: file should end in a newline

File: .travis.yml (hunk 3 above)
Comment (dfed) [inline .travis.yml]: ahhhh this makes it so we don't have to add new `xcodebuild` lines to the script below. Dig the concept

File: SocketRocket.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -13,8 +13,11 @@
 		2D4227841BB436CF000C1A6C /* SocketRocket.h in Headers */ = {isa = PBXBuildFile; fileRef = 2D4227641BB4358C000C1A6C /* SocketRocket.h */; settings = {ATTRIBUTES = (Public, ); }; };
 		2D4227851BB43734000C1A6C /* SRWebSocket.m in Sources */ = {isa = PBXBuildFile; fileRef = F6A12CD0145119B700C1D980 /* SRWebSocket.m */; };
 		2D8650261BB43B9B00534598 /* libicucore.dylib in Frameworks */ = {isa = PBXBuildFile; fileRef = F6C41C95145F7C4700641356 /* libicucore.dylib */; };
-		555E0EB21C51E56D00E6BB92 /* SocketRocket.h in Headers */ = {isa = PBXBuildFile; fileRef = 555E0EB11C51E56D00E6BB92 /* SocketRocket.h */; settings = {ATTRIBUTES = (Public, ); }; };
-		555E0EB31C51E56D00E6BB92 /* SocketRocket.h in Headers */ = {isa = PBXBuildFile; fileRef = 555E0EB11C51E56D00E6BB92 /* SocketRocket.h */; settings = {ATTRIBUTES = (Public, ); }; };
+		3345DC841C52ACD70083CCB8 /* SRWebSocket.m in Sources */ = {isa = PBXBuildFile; fileRef = F6A12CD0145119B700C1D980 /* SRWebSocket.m */; };
```
Comment (dfed) [inline SocketRocket.xcodeproj/project.pbxproj]: Curious why these changes were necessary?

File: SocketRocket.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -658,6 +726,79 @@
...
 		};
+		3345DC8E1C52ACD70083CCB8 /* Debug */ = {
+			isa = XCBuildConfiguration;
+			buildSettings = {
+				CLANG_CXX_LANGUAGE_STANDARD = "gnu++0x";
+				CLANG_CXX_LIBRARY = "libc++";
+				CLANG_ENABLE_MODULES = YES;
+				CLANG_WARN_DIRECT_OBJC_ISA_USAGE = YES_ERROR;
+				CLANG_WARN_OBJC_ROOT_CLASS = YES_ERROR;
+				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
```
Comment (dfed) [inline SocketRocket.xcodeproj/project.pbxproj]: @mtauraso @justinseanmartin do we want this?

File: SocketRocket.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -658,6 +726,79 @@
...
+		};
+		3345DC8F1C52ACD70083CCB8 /* Release */ = {
+			isa = XCBuildConfiguration;
+			buildSettings = {
+				CLANG_CXX_LANGUAGE_STANDARD = "gnu++0x";
+				CLANG_CXX_LIBRARY = "libc++";
+				CLANG_ENABLE_MODULES = YES;
+				CLANG_WARN_DIRECT_OBJC_ISA_USAGE = YES_ERROR;
+				CLANG_WARN_OBJC_ROOT_CLASS = YES_ERROR;
+				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
```
Comment (dfed) [inline SocketRocket.xcodeproj/project.pbxproj]: @mtauraso @justinseanmartin do we want this?

File: SocketRocket.xcodeproj/project.pbxproj (hunk 7 above)
Comment (justinseanmartin) [inline SocketRocket.xcodeproj/project.pbxproj]: Should be fine, same as all the other targets:
https://github.com/square/SocketRocket/blob/master/SocketRocket.xcodeproj/project.pbxproj

File: SocketRocket.xcodeproj/project.pbxproj (hunk 8 above)
Comment (dfed) [inline SocketRocket.xcodeproj/project.pbxproj]: Per the above, we're fine.

**Comment by dfed** [comment]
@hamchapman thanks for the updates! Just a few questions and we should have this merged.

File: .travis.yml (hunk 4 above)
Comment (hamchapman) [inline .travis.yml]: So it uses the new container-based infrastructure: https://docs.travis-ci.com/user/migrating-from-legacy/

File: .travis.yml (hunk 5 above)
Comment (hamchapman) [inline .travis.yml]: Will add

File: SocketRocket.xcodeproj/project.pbxproj (hunk 6 above)
Comment (hamchapman) [inline SocketRocket.xcodeproj/project.pbxproj]: Can't remember exactly but I think it was just about making sure that the headers were being setup consistently

**Comment by hamchapman** [comment]
Added the newline at the bottom of the Podfile. Is it good to go otherwise?

**Comment by dfed** [comment]
Yeah we're good. Thanks for the quick explanations and waiting for me to get back to you. Merging :)
